crates/cheatnet/src/forking/state.rs

```diff
@@ -99,9 +73,9 @@ impl BlockInfoReader for ForkStateReader {
 
                 Ok(block_info)
             }
-            Ok(MaybePendingBlockWithTxHashes::PendingBlock(pending_block)) => {
-                get_pending_block_parent(self, &pending_block)
-            }
+            Ok(MaybePendingBlockWithTxHashes::PendingBlock(_)) => Err(StateReadError(
```

Comment:
Shouldn't this be resolved on the configuration level? I think it's a bit late in code to throw an error like this (if it's not strictly for validation)

Reply:
See the conv above^